modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsConfigurationManager.java

```diff
@@ -0,0 +1,549 @@
+                .map(c -> new StatisticsColumnConfiguration(c.getName()))
+                .collect(Collectors.toList());
+
+            StatisticsObjectConfiguration newCfg = new StatisticsObjectConfiguration(target.key(), colCfgs);
+
+            try {
+                while (true) {
+                    String key = key2String(target.key());
+
+                    StatisticsObjectConfiguration oldCfg = distrMetaStorage.read(key);
+
+                    if (oldCfg != null)
+                        newCfg = StatisticsObjectConfiguration.merge(oldCfg, newCfg);
+
+                    if (distrMetaStorage.compareAndSet(key, oldCfg, newCfg))
+                        break;
+                }
+            }
+            catch (IgniteCheckedException ex) {
+                throw new IgniteSQLException("Error on get or update statistic schema", IgniteQueryErrorCode.UNKNOWN, ex);
+            }
+        }
+    }
+
+    /** */
+    public void dropStatistics(List<StatisticsTarget> targets) {
+        if (log.isDebugEnabled())
+            log.debug("Drop statistics [targets=" + targets + ']');
+
+        for (StatisticsTarget target : targets) {
+            String key = key2String(target.key());
+
+            try {
+                while (true) {
+                    StatisticsObjectConfiguration oldCfg = distrMetaStorage.read(key);
+
+                    if (oldCfg == null)
+                        break;
+
+                    StatisticsObjectConfiguration newCfg = oldCfg.dropColumns(
+                        target.columns() != null ?
+                            Arrays.stream(target.columns()).collect(Collectors.toSet()) :
+                            Collections.emptySet()
+                    );
+
+                    if (distrMetaStorage.compareAndSet(key, oldCfg, newCfg))
+                        break;
+                }
+            }
+            catch (IgniteCheckedException ex) {
+                throw new IgniteSQLException(
+                    "Error on get or update statistic schema", IgniteQueryErrorCode.UNKNOWN, ex);
+            }
+        }
+    }
+
+    /** Drop All statistics. */
+    public void dropAll() {
+        mgmtPool.submit(() -> {
+            try {
+                final List<StatisticsTarget> targetsToRemove = new ArrayList<>();
+
+                distrMetaStorage.iterate(STAT_OBJ_PREFIX, (k, v) -> {
+                        StatisticsKey statKey = ((StatisticsObjectConfiguration)v).key();
+
+                        targetsToRemove.add(new StatisticsTarget(statKey.schema(), statKey.obj()));
+                    }
+                );
+
+                dropStatistics(targetsToRemove);
+            }
+            catch (IgniteCheckedException e) {
+                log.warning("Unexpected exception drop all statistics", e);
+            }
+        });
+    }
+
+    /** */
+    private void validate(StatisticsTarget target, GridH2Table tbl) {
+        if (tbl == null) {
+            throw new IgniteSQLException(
```

Comment:
Should we use SQL exception here? User can create a table using JavaAPI and collect statistics by Java API without SQL at all.